pkg/controller/plan/kubevirt.go

```diff
@@ -696,6 +707,29 @@ func (r *KubeVirt) virtualMachine(vm *plan.VMStatus) (object *cnv.VirtualMachine
 		return
 	}
 
+	//If the VM name is not valid according to DNS1123 labeling
+	//convention it will be automatically changed.
+	var originalName string
+
+	if errs := k8svalidation.IsDNS1123Label(vm.Name); len(errs) > 0 {
+		originalName = vm.Name
+
+		generatedName := changeVmName(vm.Name, vm.ID)
+		r.mutex.Lock()
```

Comment:
why the mutex is required? maybe a comment on that?

Reply:
sure will add a comment, basically this function checks if the new generated VM name exists on the destination and if so adds the ID to the name, I've added a mutex for the case that two or more VMS executing this part at parallel and the Vm list is not updated yet and after renaming both of the names will be the same (`Vm--` and `--vm` for example both will change to `vm`) but the function my miss that point since the full VM list is not updated yet, so to avoid any issues I've added a mutex so only one VM at a time can retrieve the data.

Reply:
mutex not needed, the issue was different and solved by comparing the name with the VMS from the same plan.